test/jdk/java/awt/ScrollPane/ScrollPaneWindowsTest.java

```diff
@@ -0,0 +1,190 @@
+    Frame frame;
+    Insets paneInsets;
+    public static final Object LOCK = new Object();
+    ScrollPaneAdjustable vScroll;
+    ScrollPaneAdjustable hScroll;
+    boolean notifyReceived = false;
+    volatile int xPos = 0;
+    volatile int yPos = 0;
+
+    public static void main(String[] args) throws Exception {
+        ScrollPaneWindowsTest scrollTest = new ScrollPaneWindowsTest();
+        scrollTest.init();
+        scrollTest.start();
+    }
+
+    public void init() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            frame = new Frame("ScrollPaneWindowsTest");
+            frame.setLayout(new BorderLayout(1, 1));
+            p = new Panel();
+            p.setLayout(null);
+            p.setSize(new Dimension(800, 800));
+            sp = new ScrollPane(ScrollPane.SCROLLBARS_ALWAYS);
+            vScroll = (ScrollPaneAdjustable) sp.getVAdjustable();
+            hScroll = (ScrollPaneAdjustable) sp.getHAdjustable();
+            sp.add(p);
+            frame.add(sp);
+            frame.pack();
+            frame.setSize(400, 400);
+            frame.setLocationRelativeTo(null);
+            frame.setAlwaysOnTop(true);
+            frame.setVisible(true);
+        });
+        robot = new Robot();
+
+        vScroll.addAdjustmentListener(this);
+        hScroll.addAdjustmentListener(this);
+    }
+
+    public void start() throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                paneInsets = sp.getInsets();
+                System.out.println("Insets: right = " + paneInsets.right + " bottom =  " + paneInsets.bottom);
+            });
+
+            robot.delay(100);
+            robot.waitForIdle();
+            EventQueue.invokeAndWait(() -> {
+                xPos = sp.getLocationOnScreen().x + sp.getWidth() - paneInsets.right / 2;
+                yPos = sp.getLocationOnScreen().y + sp.getHeight() / 2;
+            });
+
+            robot.mouseMove(xPos, yPos);
+            testOneScrollbar(vScroll);
+
+            robot.delay(100);
+            robot.waitForIdle();
+
+            EventQueue.invokeAndWait(() -> {
+                xPos = sp.getLocationOnScreen().x + sp.getWidth() / 2;
+                yPos = sp.getLocationOnScreen().y + sp.getHeight() - paneInsets.bottom / 2;
+            });
+
+            robot.mouseMove(xPos, yPos);
+            testOneScrollbar(hScroll);
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+        System.out.println("Test passed. ");
+    }
+
+    public void testOneScrollbar(ScrollPaneAdjustable scroll) throws Exception {
+        //to Bottom  - right
+        robot.mousePress(InputEvent.BUTTON3_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON3_DOWN_MASK);
+        robot.delay(2000);
+        robot.waitForIdle();
+
+        notifyReceived = false;
+        synchronized (LOCK) {
```

Comment:
```suggestion
        synchronized (LOCK) {
            notifyReceived = false;
```
The `notifyReceived` must always be read and written inside a synchronised block. Other places should also be corrected.

Reply:
Updated.